react-backend/client/src/components/visitors/signup_form.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import Input from './input';
+import Select from './select';
+
+export default props => (
+  <div>
+    <form className="SignupForm" onChange={props.handleChange}>
+      <section className="Main col-9">
+        <h1>Please tell us about yourself</h1>
+        {props.error && (
+          <div className="ErrorText">
+            {props.error.map(el => <span key={el}>{el}</span>)}
+          </div>
+        )}
+        <Input question="Name" option="fullname" />
+        <Input question="Email" option="email" />
+        <Input question="Phone Number (optional)" option="phone" />
+        <Select
+          question="Gender"
+          option="gender"
+          choices={['', 'male', 'female', 'prefer not to say']}
+        />
+        <Select question="Year of Birth" option="year" choices={props.years} />
+        <button onClick={props.checkUserExists} className="Button">
+          Submit
+        </button>
+      </section>
+    </form>
+    <form className="EmailSMSCheckbox" onChange={props.handleChangeCheckbox}>
+      <aside className="Privacy col-3">
+        <h3 className="Privacy__title">
+          Why are we collecting this information?
+        </h3>
+        <p className="Privacy__text">
+          Here at XXXXXXXX, we take your privacy seriously: we will only use
```

Comment:
What's the plan with this `XXXXX...`?

Reply:
oops completely forgot about that, that's meant to be the cb's name

Reply:
You just need to differentiate between checkboxes and other input types in your change handler. I think you can inspect `e.target.type`.

Reply:
I'm gonna leave the [insert cb name] bit for a seperate pr as it will potentially involve a new query/route.

Reply:
That's fine, please create an issue for it